packages/x-date-pickers/src/DateCalendar/DayCalendar.tsx

```diff
@@ -223,6 +223,7 @@ function WrappedDay<TDate extends unknown>({
       role: 'gridcell',
       isAnimating: isMonthSwitchingAnimating,
       selectedDays,
+      'data-timestamp': utils.toJsDate(day).valueOf(),
```

Comment:
Seems like you are not using it anywhere.

Reply:
But you make a good point. I'll add a comment about how and why it is used. 😉

Reply:
Oh I forgot that the imperative API was named `dataset.*` :see_no_evil: